File: SpearoGo/Utils/Brand.swift
import SwiftUI

// ─────────────────────────────────────────────────────────────────────────────
// Brand.swift — Single source of truth for all design tokens.
//
// Usage:
//   .foregroundStyle(Brand.Colors.primary)
//   .brandFont(Brand.Typography.verdictLabel)
//   .padding(Brand.Spacing.page)
//
// All colors are backed by named color sets in Assets.xcassets/Colors/.
// Changing a color set in Xcode instantly updates every screen.
// ─────────────────────────────────────────────────────────────────────────────

enum Brand {

    // ─── Colors ──────────────────────────────────────────────────────────────

    enum Colors {
        // Base
        static let background  = Color("Background")
        static let primary     = Color("OceanBlue")     // #0077B6
        static let secondary   = Color("Teal")          // #00B4D8

        // Text
        static let textPrimary   = Color("TextPrimary")
        static let textSecondary = Color("TextSecondary")

        // Verdicts
        static let go      = Color("VerdictGo")      // #2ECC71
        static let maybe   = Color("VerdictMaybe")   // #F39C12
        static let sketchy = Color("VerdictSketchy") // #E67E22
        static let noGo    = Color("VerdictNoGo")    // #E74C3C

        // Semantic aliases — use these in views rather than raw verdict names
        static let safe      = go
        static let caution   = maybe
        static let warning   = sketchy
        static let danger    = noGo
        static let accent    = secondary

        // Convenience: verdict color from DiveScore.Verdict
        static func forVerdict(_ verdict: Verdict) -> Color {
            switch verdict {
            case .go:      return go
            case .maybe:   return maybe
            case .sketchy: return sketchy
            case .noGo:    return noGo
            }
        }
    }

    // ─── Typography ──────────────────────────────────────────────────────────

    enum Typography {
        // A type token, not a Font. Fonts are resolved per-view by `brandFont`,
        // which scales `size` against the wearer's text-size setting via
        // @ScaledMetric. `Font.system(size:)` is a FIXED size and ignores
        // Accessibility -> Larger Text entirely, which is why these were
        // unreadable for anyone who had turned it up.
        //
        // `relativeTo` picks the metric the size scales along, so a 24pt
        // verdict and an 11pt label grow at rates appropriate to their role
        // rather than all multiplying uniformly.
        struct Token {
            let size: CGFloat
            let weight: Font.Weight
            let relativeTo: Font.TextStyle
        }

        // Sizes raised 2026-08-31 after a field report that the watch UI was
        // too small to read. Nothing sits below 11pt now; the old scale
        // bottomed out at 8pt, well under the ~12pt watchOS legibility floor.
        // Prior values are noted so the hierarchy stays auditable.

        // Verdict label — "GO", "MAYBE", "SKETCHY", "NO GO"      (was 20)
        static let verdictLabel = Token(size: 24, weight: .black, relativeTo: .title2)

        // Large data values — "14", "1.2", "22"                  (was 18)
        static let dataValue = Token(size: 21, weight: .bold, relativeTo: .title3)

        // Score ring centre number                               (was 14)
        static let scoreNumber = Token(size: 17, weight: .bold, relativeTo: .headline)

        // Tide / solunar times — "14:32"                         (was 16)
        static let timeDisplay = Token(size: 19, weight: .bold, relativeTo: .title3)

        // Solunar period times                                   (was 12)
        static let periodTime = Token(size: 14, weight: .medium, relativeTo: .body)

        // Personality copy under verdict                         (was 11)
        static let personalityCopy = Token(size: 13, weight: .regular, relativeTo: .body)

        // Section headers — "CONDITIONS", "TIDES"                (was 10)
        static let sectionHeader = Token(size: 12, weight: .semibold, relativeTo: .caption)

        // Item labels — "WIND", "SWELL", "HIGH"                  (was 8)
        static let itemLabel = Token(size: 11, weight: .semibold, relativeTo: .caption2)

        // Unit labels — "kn", "m", "°C"                          (was 9)
        static let unit = Token(size: 11, weight: .regular, relativeTo: .caption2)

        // Captions — wetsuit tip, sub-labels, body copy          (was 9)
        static let caption = Token(size: 12, weight: .regular, relativeTo: .caption)
    }

    // ─── Spacing ─────────────────────────────────────────────────────────────

    enum Spacing {
        static let page:    CGFloat = 12   // view edge padding
        static let section: CGFloat = 10   // between major elements
        static let item:    CGFloat = 6    // between related items
        static let micro:   CGFloat = 2    // label-value gap
    }

    // ─── Letter spacing ──────────────────────────────────────────────────────

    enum Kerning {
        static let sectionHeader: CGFloat = 2.0
        static let itemLabel:     CGFloat = 1.0
        static let unitLabel:     CGFloat = 0.5
    }

    // ─── Corner radii ────────────────────────────────────────────────────────

    enum Radius {
        static let card:  CGFloat = 12
        static let chip:  CGFloat = 8
        static let badge: CGFloat = 6
        static let pill:  CGFloat = 100
    }

    // ─── Opacity ─────────────────────────────────────────────────────────────

    enum Opacity {
        static let ringTrack:  Double = 0.30
        static let cardFill:   Double = 0.04
        static let borderLine: Double = 0.08
        static let disabled:   Double = 0.35
    }

    // ─── Score ring dimensions ────────────────────────────────────────────────

    enum Ring {
        static let size:        CGFloat = 58
        static let strokeWidth: CGFloat = 5
        static let radius:      CGFloat = 24
    }
}

File: wear/app/src/main/java/com/spearotracker/spearogo/ui/theme/Brand.kt
package com.spearotracker.spearogo.ui.theme

import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.spearotracker.spearogo.models.Verdict

object Brand {

    object Colors {
        // Base
        val background = Color(0xFF000000)
        val primary = Color(0xFF0077B6)      // OceanBlue
        val secondary = Color(0xFF00B4D8)    // Teal

        // Text
        val textPrimary = Color(0xFFFFFFFF)
        val textSecondary = Color(0xFF6B7D8E)

        // Verdicts
        val go = Color(0xFF2ECC71)
        val maybe = Color(0xFFF39C12)
        val sketchy = Color(0xFFE67E22)
        val noGo = Color(0xFFE74C3C)

        // Semantic aliases
        val safe = go
        val caution = maybe
        val warning = sketchy
        val danger = noGo
        val accent = secondary

        fun forVerdict(verdict: Verdict): Color = when (verdict) {
            Verdict.GO -> go
            Verdict.MAYBE -> maybe
            Verdict.SKETCHY -> sketchy
            Verdict.NO_GO -> noGo
        }
    }

    object Typography {
        // Sizes raised 2026-08-31 after a field report that the watch UI was too
        // small to read. Nothing sits below 11.sp now; the old scale bottomed out
        // at 8.sp, well under the Wear Material legibility floor. Prior values
        // are noted so the hierarchy stays auditable, and these must stay in
        // lockstep with SpearoGo/Utils/Brand.swift — the two scales are the same
        // design and have already drifted into the same bug once.
        //
        // These are `sp`, not `dp`, so they follow the wearer's Settings ->
        // Display -> Font size. SpearoGoTheme bounds how far that goes.
        val verdictLabel = TextStyle(fontSize = 24.sp, fontWeight = FontWeight.Black)        // was 20
        val dataValue = TextStyle(fontSize = 21.sp, fontWeight = FontWeight.Bold)            // was 18
        val scoreNumber = TextStyle(fontSize = 17.sp, fontWeight = FontWeight.Bold)          // was 14
        val timeDisplay = TextStyle(fontSize = 19.sp, fontWeight = FontWeight.Bold)          // was 16
        val periodTime = TextStyle(fontSize = 14.sp, fontWeight = FontWeight.Medium)         // was 12
        val personalityCopy = TextStyle(fontSize = 13.sp, fontWeight = FontWeight.Normal)    // was 11
        val sectionHeader = TextStyle(fontSize = 12.sp, fontWeight = FontWeight.SemiBold, letterSpacing = 2.sp)  // was 10
        val itemLabel = TextStyle(fontSize = 11.sp, fontWeight = FontWeight.SemiBold, letterSpacing = 1.sp)      // was 8
        val unit = TextStyle(fontSize = 11.sp, fontWeight = FontWeight.Normal)               // was 9
        val caption = TextStyle(fontSize = 12.sp, fontWeight = FontWeight.Normal)            // was 9
    }

    object Spacing {
        val page = 12.dp
        val section = 10.dp
        val item = 6.dp
        val micro = 2.dp
    }

    object Kerning {
        val sectionHeader = 2.sp
        val itemLabel = 1.sp
        val unitLabel = 0.5.sp
    }

    object Radius {
        val card = 12.dp
        val chip = 8.dp
        val badge = 6.dp
        val pill = 100.dp
    }

    object Opacity {
        const val ringTrack = 0.30f
        const val cardFill = 0.04f
        const val borderLine = 0.08f
        const val disabled = 0.35f
    }

    object Ring {
        val size = 58.dp
        val strokeWidth = 5.dp
    }
}
